tests/trainer/test_trainer.py

```diff
@@ -1760,6 +1760,36 @@ def test_model_in_correct_mode_during_stages(tmpdir, accelerator, num_processes)
     trainer.predict(model, model.val_dataloader())
 
 
+class TrainerStagesErrorsModel(BoringModel):
+    def on_train_start(self) -> None:
+        raise BaseException("Error during train")
+
+    def on_validation_start(self) -> None:
+        raise BaseException("Error during validation")
+
+    def on_test_start(self) -> None:
+        raise BaseException("Error during test")
+
+    def on_predict_start(self) -> None:
+        raise BaseException("Error during predict")
+
```

Comment:
you may want to define a custom exception class so the error heck in the test is specific. otherwise any exception could occur, and the test would pass

Reply:
good call! Done